### PR TITLE
drivers: ncp5623: Fix error check

### PR DESCRIPTION
**Pull request by finikorg**
num_colors cannot be not equal to different values at the same time.

### CODE REVIEW

File: drivers/led/ncp5623.c [hunk 1]
```diff
@@ -149,7 +149,7 @@ static int ncp5623_led_init(const struct device *dev)
 			return -ENODEV;
 		}
 
-		if (led_info->num_colors != 3 || led_info->num_colors != 1) {
+		if (led_info->num_colors != 3 && led_info->num_colors != 1) {
```
Comment (simonguinot) [inline drivers/led/ncp5623.c:152]: Thanks for spotting this.

In addition I think the code should rather be:

```
if (led_info->num_colors != NCP5623_CHANNEL_COUNT) {
                LOG_ERR("%s: invalid number of colors %d (must be %d when defining a single LED)",
                                  dev->name, led_info->num_colors, NCP5623_CHANNEL_COUNT);
}
```

@PetervdPerk-NXP, please can you have a look ?

Comment (PetervdPerk-NXP) [reply drivers/led/ncp5623.c:152]: It was for the weird edge where one would be attaching one singe color led to the ncp5623.
@simonguinot suggestion to just enforce 3 colors in single led rgb mode would make more sense.

Comment (finikorg) [reply drivers/led/ncp5623.c]: Fixed proposed way.

**Review by simonguinot — APPROVED** [review]
Thanks for taking care @finikorg !

**Review by PetervdPerk-NXP — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by jhedberg** [comment]
> Shouldn't we return any errno in that case?

Yeah, at least to me it seems like `return -EINVAL` would be the right thing (and consistent with how the function is dealing with other similar errors).

File: drivers/led/ncp5623.c [hunk 2]
```diff
@@ -149,9 +149,9 @@ static int ncp5623_led_init(const struct device *dev)
 			return -ENODEV;
 		}
 
-		if (led_info->num_colors != 3 || led_info->num_colors != 1) {
-			LOG_ERR("%s: invalid number of colors %d (must be %d or 1)", dev->name,
-				led_info->num_colors, NCP5623_CHANNEL_COUNT);
+		if (led_info->num_colors != NCP5623_CHANNEL_COUNT) {
+			LOG_ERR("%s: invalid number of colors %d (must be %d with a single LED)",
+				dev->name, led_info->num_colors, NCP5623_CHANNEL_COUNT);
```
Comment (simonguinot) [inline drivers/led/ncp5623.c:154]: Please can you add the missing return as well ?

**Review by simonguinot — CHANGES_REQUESTED** [review]
Let's prevent this to be merged before `return` is added.

**Comment by simonguinot** [comment]
> Shouldn't we return any errno in that case?

Well spotted !

File: drivers/led/ncp5623.c (hunk 2 above)
Comment (finikorg) [reply drivers/led/ncp5623.c:154]: done

**Review by fabiobaltieri — APPROVED** [review]

**Review by bbilas — APPROVED** [review]

**Review by simonguinot — APPROVED** [review]
Thanks !

**Review by jhedberg — APPROVED** [review]
Looks good, however since we're at the last stages of the release, it's unlikely this will get merged before 3.5 (would require that it's fixing a release blocker issue)

**Pull request merged**